### PR TITLE
add SOPI2IRetrieval dataset/task

### PR DESCRIPTION
**Pull request by Jamie-Stirling**
* Add SOPI2IRetrieval task: 22k online products with around 4-5 images each, task is to re-identify product by a different image
* Add corresponding results for CLIP ViT patch 32

## Checklist

- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`.

### CODE REVIEW

File: mteb/models/blip2_models.py [hunk 1]
```diff
@@ -13,7 +13,11 @@
 
 
 def blip2_loader(**kwargs):
+<<<<<<< HEAD
```
Comment (KennethEnevoldsen) [inline mteb/models/blip2_models.py:16]: Merge error

Comment (Jamie-Stirling) [reply mteb/models/blip2_models.py]: thanks, this is resolved now

File: mteb/tasks/Image/Any2AnyRetrieval/eng/SOPI2IRetrieval.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+from mteb.abstasks.Image.AbsTaskAny2AnyRetrieval import AbsTaskAny2AnyRetrieval
+from mteb.abstasks.TaskMetadata import TaskMetadata
+
+
+class SOPI2IRetrieval(AbsTaskAny2AnyRetrieval):
+    metadata = TaskMetadata(
+        name="SOPI2IRetrieval",
+        description="Retrieve product photos of 22634 online products.",
+        reference="https://www.florian-schroff.de/publications/CUB-200.pdf",
```
Comment (isaac-chung) [inline mteb/tasks/Image/Any2AnyRetrieval/eng/SOPI2IRetrieval.py:11]: Different link instead of CUB200?

**Review by isaac-chung — COMMENTED** [review]
Thanks! Looks good at a glance. One suggestion. Didn't get to check the results.

**Review by gowitheflow-1998 — APPROVED** [review]

**Pull request merged**